File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/fight/GameFightResumeSlaveInfo.scala





















// Generated on 05/03/2013 10:35:04
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait GameFightResumeSlaveInfo extends TypeDeserializer {
  val typeId = 364

  protected trait Type extends BaseType with Serializable {
    val typeId = GameFightResumeSlaveInfo.typeId

    val slaveId: Int
    val spellCooldowns: Seq[GameFightSpellCooldown.TypeImpl]
    val summonCount: Byte
    val bombCount: Byte

    def serialize(buf: Buffer) {
      buf.writeInt(slaveId)
      buf.writeUShort(spellCooldowns.length)
      for (entry <- spellCooldowns) {
        entry.serialize(buf)
      }
      buf.writeByte(summonCount)
      buf.writeByte(bombCount)
    }

  }

  protected def create(slaveId: Int, spellCooldowns: Seq[GameFightSpellCooldown.TypeImpl], summonCount: Byte, bombCount: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val slaveId = buf.readInt()
    val spellCooldowns = {
      val builder = collection.immutable.Seq.newBuilder[GameFightSpellCooldown.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += GameFightSpellCooldown.deserialize(buf)
      }
      builder.result()
    }
    val summonCount = buf.readByte()
    val bombCount = buf.readByte()

    create(slaveId, spellCooldowns, summonCount, bombCount)
  }

}

object GameFightResumeSlaveInfo extends GameFightResumeSlaveInfo {
  case class TypeImpl(slaveId: Int, spellCooldowns: Seq[GameFightSpellCooldown.TypeImpl], summonCount: Byte, bombCount: Byte) extends Type
  override def create(slaveId: Int, spellCooldowns: Seq[GameFightSpellCooldown.TypeImpl], summonCount: Byte, bombCount: Byte) = TypeImpl(slaveId, spellCooldowns, summonCount, bombCount)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(slaveId: Int, spellCooldowns: Seq[GameFightSpellCooldown.TypeImpl], summonCount: Byte, bombCount: Byte) = create(slaveId, spellCooldowns, summonCount, bombCount)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/GameFightResumeSlaveInfo.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameFightResumeSlaveInfo extends NetworkType {
    public static final short TYPE_ID = 364;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int slaveId;
    public GameFightSpellCooldown[] spellCooldowns;
    public byte summonCount;
    public byte bombCount;
    
    public GameFightResumeSlaveInfo() { }
    
    public GameFightResumeSlaveInfo(int slaveId, GameFightSpellCooldown[] spellCooldowns, byte summonCount, byte bombCount) {
        this.slaveId = slaveId;
        this.spellCooldowns = spellCooldowns;
        this.summonCount = summonCount;
        this.bombCount = bombCount;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(slaveId);
        buf.writeUShort(spellCooldowns.length);
        for (GameFightSpellCooldown entry : spellCooldowns) {
            entry.serialize(buf);
        }
        buf.writeByte(summonCount);
        buf.writeByte(bombCount);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        slaveId = buf.readInt();
        int limit = buf.readUShort();
        spellCooldowns = new GameFightSpellCooldown[limit];
        for (int i = 0; i < limit; i++) {
            spellCooldowns[i] = new GameFightSpellCooldown();
            spellCooldowns[i].deserialize(buf);
        }
        summonCount = buf.readByte();
        if (summonCount < 0)
            throw new RuntimeException("Forbidden value on summonCount = " + summonCount + ", it doesn't respect the following condition : summonCount < 0");
        bombCount = buf.readByte();
        if (bombCount < 0)
            throw new RuntimeException("Forbidden value on bombCount = " + bombCount + ", it doesn't respect the following condition : bombCount < 0");
    }
    
}
